Combine redundancy funcs

aelanman opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
Replaces the `UVData.get_antenna_redundancies` and `UVData.get_baseline_redundancies` methods with  `UVData.get_redundancies`. The new method has a keyword that chooses which behavior to use.
Also changes all redundancy finder methods to always return _four_ items -- redundant groups, vector bin centers, lengths, and conjugate list. In most cases, the conjugate list is None. This allows for more consistent behavior when called.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
The names `get_antenna_redundancies` and `get_baseline_redundancies` were not sufficiently descriptive. One function returns baseline redundancies for the entire `telescope`, as defined by the antenna positions array, while the other finds redundancies among the baselines present in the data. This change replaces both functions with one, with an option `use_antpos` to choose between the previous function options.

<!--- If it fixes an open issue, please link to the issue here. If this PR closes an issue, put the word 'closes' before the issue link to auto-close the issue when the PR is merged. -->
closes #645 

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation change (documentation changes only)
- [ ] Version change
- [ ] Build or continuous integration change


## Checklist:
<!--- You may remove the checklists that don't apply to your change type(s) or just leave them empty -->
<!--- Go over all the following points, and replace the space with an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have read the [contribution guide](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/master/.github/CONTRIBUTING.md).
- [x] My code follows the code style of this project.

Breaking change checklist:  
- [x] I have updated the docstrings associated with my change using the [numpy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html).
- [x] I have updated the tutorial to reflect my changes (if appropriate).
- [ ] My change includes backwards compatibility and deprecation warnings (if possible).
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests pass in both python 2 and python 3.
- [x] I have updated the [CHANGELOG](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/master/CHANGELOG.md).

bhazelton commented on CHANGELOG.md:
```diff
@@ -3,6 +3,11 @@ All notable changes to this project will be documented in this file.
+- All redundancy finder methods will return four items --- redundant groups, bin centers, lengths, and conjugates.
+- UVData.get_antenna_redundancies will no longer automatically conjugate baselines.
+- UVData.get_baseline_redundancies and UVData.get_antenna_redundancies have been combined.
+
```
should add a deprecated section to call out the deprecated methods.

aelanman replied:
Done

bhazelton commented on CHANGELOG.md:
```diff
@@ -3,6 +3,11 @@ All notable changes to this project will be documented in this file.
+### Changed
+- All redundancy finder methods will return four items --- redundant groups, bin centers, lengths, and conjugates.
+- UVData.get_antenna_redundancies will no longer automatically conjugate baselines.
+- UVData.get_baseline_redundancies and UVData.get_antenna_redundancies have been combined.
```
The wording in these two lines is weird -- you're calling out a change to a method that is being deprecated and the new method isn't mentioned.

aelanman replied:
I've moved this to the Added section and rephrased it to say that the new function replaces the old two.

bhazelton commented on pyuvdata/utils.py:
```diff
@@ -1156,7 +1156,7 @@ def get_baseline_redundancies(baselines, baseline_vecs, tol=1.0, with_conjugates
     lens = np.sqrt(np.sum(vec_bin_centers**2, axis=1))
 
-    return bl_gps, vec_bin_centers, lens
+    return bl_gps, vec_bin_centers, lens, None
```
I'm a bit confused by returning `None` here. I'm not sure it's a better interface to ensure that we return the same number of things by returning a `None`. In fact, this recently led to confusion in the beam interp functions (see issue #622). @jpober what do you think?

aelanman commented on pyuvdata/uvdata.py:
```diff
@@ -4449,47 +4453,54 @@ def get_antenna_redundancies(self, tol=1.0, include_autos=True,
+        """
+
+        warnings.warn("UVData.get_antenna_redundancies has been replaced with get_redundancies.",
+                      DeprecationWarning)
```
I've updated both warnings as suggested.

aelanman commented on pyuvdata/uvdata.py:
```diff
@@ -4423,23 +4423,27 @@ def _calc_single_integration_time(self):
-        Get redundant baselines to a given tolerance from antenna positions.
+        Get redundant baselines to a given tolerance.
 
         Finds all possible redundant baselines (antenna pairs) not just those with data.
```
Yes, I missed that one. Thanks! It's removed now.

jpober commented:
I agree that returning None by default is awkward.  Maybe you can make it return 3 items if include_conjugates is False and return 4 items if it's True?  If use_antpos and include_conjugates are both True, then it can return None for the 4th item.

aelanman commented:
That makes sense. A user calling with `include_conjugates` should expect a fourth returned item.  I've changed the behavior and updated the docstring/changelog accordingly.

bhazelton commented:
@aelanman sorry, but this needs to be rebased again for the changelog. I can do it if you like.

aelanman commented:
@bhazelton No problem. I also found a typo in an earlier commit message.
